### PR TITLE
pull request #5 from tenyiting/remember_me

### PR DESCRIPTION
**Pull request by tenyiting**
Ready

### CODE REVIEW

**Comment by tenyiting** [comment]
ready

File: app/models/user.rb [hunk 1]
```diff
@@ -1,24 +1,43 @@
...
+  def remember
+    self.remember_token = User.new_token
+    update_attribute :remember_digest, User.digest(remember_token)
+  end
+
+  def forget
+    update_attribute :remember_digest, nil
+  end
 
   private
```
Comment (huongnt-2545) [inline app/models/user.rb:23]: move the private methods to after defined class method
ex:
```suggestion
  class << self
    def new_token
      SecureRandom.urlsafe_base64
    end
    ...
  end
  
  private
  
  def downcase_email
    self.email.downcase!
  end
```

**Review by huongnt-2545 — COMMENTED** [review]
commented
You need to create new commit for each branch,
you are using the merged commit from previous branch
Please check

**Comment by tenyiting** [comment]
Ready

File: app/models/user.rb [hunk 2]
```diff
@@ -1,24 +1,30 @@
...
   validates :name, presence: true
   VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i
   validates :email, presence: true, length: {maximum: 255}, format: {with: VALID_EMAIL_REGEX}, uniqueness: true
 
-  private
+  has_secure_password
 
   def downcase_email
     self.email.downcase!
   end
```
Comment (anlvq-1058) [inline app/models/user.rb:12]: put your downcase_email under private

**Review by anlvq-1058 — COMMENTED** [review]
commented
You need to create new commit for each branch,
you are using the merged commit from previous branch
Please check

**Comment by tenyiting** [comment]
Ready

**Comment by tenyiting** [comment]
Ready

**Review by anlvq-1058 — COMMENTED** [review]
merged

**Pull request merged**